lib/pbench/server/api/resources/users_api.py

```diff
@@ -269,14 +268,24 @@ def post(self):
                         "message": "failure message"
                     }
         """
-        auth_token = self.auth.token_auth.get_auth()["token"]
-        user = self.auth.token_auth.current_user()
+        auth_token = self.auth.get_auth_token(self.logger)
+        user = Auth.verify_auth(auth_token=auth_token)
+
+        # "None" user represents that either the token is not present in our database or it is an expired token.
+        # Expired token is already deleted by now if we reach here.
+        # In either case we will return suceess to a client, since the same token can not be used again.
 
         try:
             ActiveTokens.delete(auth_token)
```

Comment:
Is an exception raised when a previously deleted `auth_token` is used?

Reply:
No, deleting a row that doesn't exists is not an error or exception.

Reply:
So why are we executing a database operation to remove the token if we found that it is not associated with the user?

Should this be:

```
if user:
    try:
        ActiveTokens.delete(auth_token)
    except Exception:
        self.logger.exception("Exception ...")
        abort(500, ...)
    else:
        self.logger.debug("User {} logged out", user.username)
else:
    self.logger.debug("User logout with invalid or expired token")

return "", 200
```

Reply:
Agreed; it may be a silent no-op to SQLAlchemy, but it's going to be a constant semantic confusion to anyone reading the code and there's no point in calling SQLAlchemy for an expensive no-op when we know there's nothing to do.

Reply:
That is a good point